src/gobexport/csv_inspector.py

```diff
@@ -108,10 +117,25 @@ def _check_lengths(self, columns: list[str]):
         :param columns:
         :return:
         """
+        cols = self.cols
+        result = {}
+        minlen_str = self.cols_min_len(len(columns))
+        maxlen_str = self.cols_max_len(len(columns))
+
         for i, column in enumerate(columns):
-            column_len = len(column)
-            self.cols[f"minlength_col_{i + 1}"] = min(column_len, self.cols.get(f"minlength_col_{i + 1}", column_len))
-            self.cols[f"maxlength_col_{i + 1}"] = max(column_len, self.cols.get(f"maxlength_col_{i + 1}", column_len))
+            length = len(column)
+
+            try:
+                if length < cols[minlen_str[i]]:
```

Comment:
Was this really a bottleneck, or is this a "just in case" kind of thing?

The O of this function doesn't change. All you're doing is replacing two dict assignments with other logic (cache lookup, possible KeyError handling, dict update). At first sight I don't see how this is considerably faster; but the code gets more complicated and harder to read though. So unless there's a very good reason to choose this implementation over the existing one, I'd keep the existing one.

Have you timed this?